src/Platform/Microsoft.Testing.Platform/Hosts/CommonTestHost.cs

```diff
@@ -56,11 +56,56 @@ public async Task<int> RunAsync()
             await DisposeServiceProviderAsync(ServiceProvider, isProcessShutdown: true);
             await DisposeHelper.DisposeAsync(ServiceProvider.GetService<FileLoggerProvider>());
             await DisposeHelper.DisposeAsync(ServiceProvider.GetTestApplicationCancellationTokenSource());
+            await DisposeHelper.DisposeAsync(dotnetTestPipeClient);
         }
 
         return exitCode;
     }
 
+    private async Task<int> RunTestAppAsync(CancellationToken testApplicationCancellationToken)
+    {
+        if (RunTestApplicationLifeCycleCallbacks)
+        {
+            // Get the test application lifecycle callbacks to be able to call the before run
+            foreach (ITestApplicationLifecycleCallbacks testApplicationLifecycleCallbacks in ServiceProvider.GetServicesInternal<ITestApplicationLifecycleCallbacks>())
+            {
+                await testApplicationLifecycleCallbacks.BeforeRunAsync(testApplicationCancellationToken);
+            }
+        }
+
+        int exitCode = await InternalRunAsync();
+
+        if (RunTestApplicationLifeCycleCallbacks)
+        {
+            foreach (ITestApplicationLifecycleCallbacks testApplicationLifecycleCallbacks in ServiceProvider.GetServicesInternal<ITestApplicationLifecycleCallbacks>())
+            {
+                await testApplicationLifecycleCallbacks.AfterRunAsync(exitCode, testApplicationCancellationToken);
+                await DisposeHelper.DisposeAsync(testApplicationLifecycleCallbacks);
+            }
+        }
+
+        return exitCode;
+    }
+
+    private async Task<bool> DoHandshakeAsync(NamedPipeClient dotnetTestPipeClient, ITestApplicationModuleInfo? testApplicationModuleInfo, CancellationToken testApplicationCancellationToken)
+    {
+        HandshakeInfo handshakeInfo = new(new Dictionary<string, string>()
+        {
+            { HandshakeInfoPropertyNames.PID, ServiceProvider.GetProcessHandler().GetCurrentProcess().Id.ToString(CultureInfo.InvariantCulture) },
+            { HandshakeInfoPropertyNames.Architecture, RuntimeInformation.OSArchitecture.ToString() },
+            { HandshakeInfoPropertyNames.Framework, RuntimeInformation.FrameworkDescription },
+            { HandshakeInfoPropertyNames.OS, RuntimeInformation.OSDescription },
+            { HandshakeInfoPropertyNames.ProtocolVersion, ProtocolConstants.Version },
+            { HandshakeInfoPropertyNames.HostType, this.GetType().Name },
+            { HandshakeInfoPropertyNames.ModulePath, testApplicationModuleInfo?.GetCurrentTestApplicationFullPath() ?? string.Empty },
+        });
+
+        HandshakeInfo response = await dotnetTestPipeClient.RequestReplyAsync<HandshakeInfo, HandshakeInfo>(handshakeInfo, testApplicationCancellationToken);
+
+        return response.Properties.TryGetValue(HandshakeInfoPropertyNames.ProtocolVersion, out string? protocolVersion) &&
```

Comment:
Check should follow the SemVer if major is compatible we're fine. We can for now go on with this but we should fix it soon.

Reply:
Okay, I'll do it in the next PR.